cmd/agent/app/agent.go

```diff
@@ -29,20 +30,22 @@ import (
 
 // Agent is a composition of all services / components
 type Agent struct {
+	proxy      CollectorProxy
 	processors []processors.Processor
 	httpServer *http.Server
 	httpAddr   atomic.Value // string, set once agent starts listening
 	logger     *zap.Logger
-	closer     io.Closer
 }
 
 // NewAgent creates the new Agent.
 func NewAgent(
+	proxy CollectorProxy,
```

Comment:
Why do we need Agent to be responsible for closing CollectorProxy if it didn't create it? There may be other services using that connection beyond the main agent's services.

Reply:
True. It makes the all-in-one a bit more complicated for now, but it is indeed the right thing to do.